release notes/v0.37.0.md

```diff
@@ -0,0 +1,71 @@
+
+**Docs**: [Using file output](https://k6.io/docs/using-k6/options/#file)
+
+### Added experimental basic event loop ([#882](https://github.com/grafana/k6/issues/882))
+
+We added basic event loop functionality ([#2228](https://github.com/grafana/k6/pull/2228) and [#2373](https://github.com/grafana/k6/pull/2373)). This is a first step and currently isn't used by anything, but is now available to extensions ([an example](https://github.com/MStoykov/xk6-events)). In following releases this will be used by k6 and will mean that some APIs will start having callbacks or return Promises. But first, it's likely that it will be used in some experimental extensions.
+
+:information_source: If you are an extension developer, please use it and give your feedback. But take into consideration that it's likely that the current Go API may change.
+
+## Extensions
+
+`v0.37.0` finalizes ([#2376](https://github.com/grafana/k6/pull/2376)) the switching of our internal modules to [a new Go/JavaScript module API](https://k6.io/docs/extensions/guides/create-an-extension/#advanced-javascript-extension).  Lastly, the internal gRPC module was refactored ([#2365](https://github.com/grafana/k6/pull/2365)).
+
+:warning: It's important to highlight that the old API (e.g. methods like `context.WithRuntime`, `common.Bind` and others [#2384](https://github.com/grafana/k6/pull/2384)) is deprecated and will be removed in the next k6 release (`v0.38.0`). For this release, every extension that isn't using the new API will get a warning message like this:
+
+```sh
+WARN[0000] Module 'k6/x/sql' is using deprecated APIs that will be removed in k6 v0.38.0, for more details on how to update it see https://k6.io/docs/extensions/guides/create-an-extension/#advanced-javascript-extension
+```
+
+We did migrations for some xk6 extensions (see connected issues to the task [#2344](https://github.com/grafana/k6/issues/2344)). The pull requests can serve as examples on how to transition your extension to the new API.
+
+## Docker Repository
+
+We migrated our Docker Hub repository from `loadimpact/k6` to [grafana/k6](https://hub.docker.com/r/grafana/k6) ([#2377](https://github.com/grafana/k6/pull/2377)).
+
+```sh
+docker run -i grafana/k6 run - <script.js
+```
+
+We will continue publishing our docker image releases as both `loadimpact/k6` and `grafana/k6` for several more releases, but if you use the old one in your local or CI environments, please plan the migration.
+
+## Enhancements and UX improvements
+
+- We continued work on the source map feature: add samples ([#2339](https://github.com/grafana/k6/pull/2339)) and improve loading and parsing ([#2355](https://github.com/grafana/k6/pull/2355)).
+- Updated installation from source instructions ([#2359](https://github.com/grafana/k6/pull/2359)). Thanks, @[karitham](https://github.com/Karitham), for making the change.
+- Introduced Go-based thresholds parsing ([#2400](https://github.com/grafana/k6/pull/2400)).
```

Comment:
Hmm this should probably be mentioned as a slight breaking change, right? We are validating thresholds a bit now, so some tests that previously "worked", or rather, silently failed, will now fail not so silently, right @oleiade?

Reply:
Any suggestions how to phrase it then? Because "Introduced Go-based thresholds parsing" will mean absolutely nothing to the vast majority of k6 users. Normal users didn't even know thresholds were implemented with JS, that was an internal implementation detail that was never advertised, so "Go-based" is at best confusing

Reply:
```suggestion
- Introduced stricter thresholds parsing ([#2400](https://github.com/grafana/k6/pull/2400)). Which then let k6 return an error message on thresholds that wouldn't have been evaluated correctly, instead of just silently ignoring them.
```


Given that in reality probably nobody cared about the fact it was in js - I would argue saying that it's "stricter" which is true is the most accurate. We might also add that this also makes it possible to give *any* feedback on the correctness of a threshold.

Reply:
We might also want to document the threshold syntax in the docs ? WDYT @oleiade